File: CS/TreeObject.cs
using System;
using DevExpress.Xpo;

public class TreeObject : XPObject {
    public TreeObject(Session session) : base(session) { }
    private string _Name;
    public string Name {
        get { return _Name; }
        set { SetPropertyValue("Name", ref _Name, value); }
    }
    private TreeObject _Parent;
    [Association("Root-Children")]
    public TreeObject Parent {
        get { return _Parent; }
        set { SetPropertyValue("Parent", ref _Parent, value); }
    }
    [Association("Root-Children")]
    public XPCollection<TreeObject> Children {
        get {
            return GetCollection<TreeObject>("Children");
        }
    }
}

File: VB/TreeObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Public Class TreeObject
	Inherits XPObject
	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub
	Private _Name As String
	Public Property Name() As String
		Get
			Return _Name
		End Get
		Set(ByVal value As String)
			SetPropertyValue("Name", _Name, value)
		End Set
	End Property
	Private _Parent As TreeObject
	<Association("Root-Children")> _
	Public Property Parent() As TreeObject
		Get
			Return _Parent
		End Get
		Set(ByVal value As TreeObject)
			SetPropertyValue("Parent", _Parent, value)
		End Set
	End Property
	<Association("Root-Children")> _
	Public ReadOnly Property Children() As XPCollection(Of TreeObject)
		Get
			Return GetCollection(Of TreeObject)("Children")
		End Get
	End Property
End Class